analyzer/pubclient/pubclient.go

```diff
@@ -0,0 +1,92 @@
+package pubclient
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"syscall"
+	"time"
+
+	coreCommon "github.com/oasisprotocol/oasis-core/go/common"
+)
+
+var permittedNetworks = map[string]bool{
+	"tcp4": true,
+	"tcp6": true,
+}
+
+type NotPermittedError struct {
+	// Note: .error is the implementation of .Error, .Unwrap etc. It is not
+	// in the Unwrap chain. Use something like
+	// `EVMDeterministicError{fmt.Errorf("...: %w", err)}` to set up an
+	// instance with `err` in the Unwrap chain.
+	error
+}
+
+func (err NotPermittedError) Is(target error) bool {
+	if _, ok := target.(NotPermittedError); ok {
+		return true
+	}
+	return false
+}
+
+// Client is an *http.Client that permits HTTP(S) connections to hosts that
+// oasis-core considers "likely to be globally reachable" on the default
+// HTTP(S) ports and unreserved ports.
+var Client = &http.Client{
```

Comment:
Could consider not exposing this and require the use through `GetWithContext`? 

but no strong opinions about it.

Reply:
sure, GetWithContext ought to be enough to get us started

Reply:
changed to private